packages/vm/lib/evm/evm.ts

```diff
@@ -297,12 +331,15 @@ export default class EVM {
       result.gasUsed = totalGas
     } else {
       if (this._vm._common.gteHardfork('homestead')) {
+        debug(`Not enough gas or code size not allowed (>= Homestead)`)
         result = { ...result, ...OOGResult(message.gasLimit) }
       } else {
         // we are in Frontier
+        debug(`Not enough gas or code size not allowed (Frontier)`)
         if (totalGas.sub(returnFee).lte(message.gasLimit)) {
           // we cannot pay the code deposit fee (but the deposit code actually did run)
-          result = { ...result, ...COOGResult(totalGas.sub(returnFee)) }
+          //result = { ...result, ...COOGResult(totalGas.sub(returnFee)) }
+          result.gasUsed = totalGas.sub(returnFee)
```

Comment:
Out of curiosity - this is a simplification e.g the result is already formatted as COOGResult? 

Should [COOGResult][1] be removed in this PR?

[1]: https://github.com/ethereumjs/ethereumjs-monorepo/blob/80093e6f650427d3cf25d15f2b12fd7e3948c803/packages/vm/lib/evm/evm.ts#L78-L92

Reply:
Oh oh oh, this actually sneaked in along doing debugging work for #1076, this shouldn't be here. 😏

@jochem-brouwer just to make you aware, can you please take care of this along working on #1081 respectively along some rebase since you are touching this code anyhow? Thanks!